test/extended/pods/liveness_override.go

```diff
@@ -0,0 +1,49 @@
+package pods
+
+import (
+	"strconv"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	"github.com/openshift/library-go/pkg/build/naming"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	e2ecommon "k8s.io/kubernetes/test/e2e/common"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+var _ = g.Describe("[sig-node]", func() {
+	defer g.GinkgoRecover()
+
+	f := framework.NewDefaultFramework("liveness-probe-override")
+
+	// upstream e2e will test normal grace period on shutdown
+	g.It("should override timeoutGracePeriodSeconds when annotation is set", func() {
+		g.By("creating the pod")
+		podName := naming.GetPodName("pod-liveness-override", string(uuid.NewUUID()))
+		pod := e2epod.NewAgnhostPod(f.Namespace.Name, podName, nil, nil, nil, "bash", "-c", "sleep 1000")
+		gracePeriod := int64(60)
```

Comment:
Set the pod level much much higher so it's super obvious when it fails - 5 or 10m?

Reply:
The test will timeout after 30s but I can do that. https://github.com/openshift/origin/pull/25828/files#diff-eceac8f75c3d7d1cae6db4c354feb7a2818a29d982421378f9da3a254bdd6075R47